libraries/default.rb

```diff
@@ -0,0 +1,68 @@
+def collect_github_repositories(github_token, org)
+  if ::ObjectSpace.const_defined?('Chef')
+    # Chef exists! Lets make sure Octokit is installed.
+    chef_gem 'octokit'
+  end
```

Comment:
Ok. So what makes the most sense to me would be moving the github functions to its own `lib/github.rb` and then installing the chef_gem first, then requiring 'octokit'. Just make sure the chef_gem is set with complie_time true so its installed before hand.